constants.go

```diff
@@ -321,6 +321,12 @@ const (
 	// Region is AWS region parameter
 	Region = "region"
 
+	//Endpoint is an optional Host for non-AWS S3
+	Endpoint = "endpoint"
+
+	//DisableSSL is an optional switch to use http
+	DisableSSL = "disablessl"
```

Comment:
would it be possible to omit this flag?

Reply:
I will describe the use case for that. 

In an above comment (https://github.com/gravitational/teleport/pull/3057#issuecomment-539541503) I've asked if Teleport is capable in encrypting the recorded session, however it relies on the s3 buckets functionality together with KMS to achieve that. Some cloud providers don't have that (MS Azure for example has another technology, Key Vault). Therefore we have developed our own encryption proxy that is deployed in the same pod as the auth server in a side container, it is targetted only over localhost: `s3://bucket-name/path/to/dir?endpoint=127.0.0.1:8080`. As it is only using `localhost` communication cannot be sniffed.

The flag also makes testing that much easier, its the same functionality you provide wit the etcd related `insecure` flag (https://gravitational.com/teleport/docs/admin-guide/#using-etcd) or the overall `--insecure` and `--insecure-no-tls` flags. Maybe renaming that parameter to `insecurehttp` or something like that would indicate to not use it in production environments?

Reply:
@klizhentas is this use case description sufficient for you? Can we proceed? :)

Reply:
There are a couple of other things we'd have to do before we proceed:

* `DisableSSL` -> `InsecureDisableTLS` SSL is a protocol that is no longer used, so this term is inaccurate.
* Tests. This pull request should add appropriate tests
* Scheme. Adding query parameters to `s3://` scheme seems like a viable solution to me, however I wonder if we can come up with something more elegant here. Still scratching my head about this one though.

Thanks for your patience.

Reply:
Then I would opt for `--insecure-no-tls` to be compliant with the other parameter flag. I will also add appropriate tests to the PR.

Reply:
The test suite also assumes a) running solely on AWS and b) there is only a TLS connection. Not sure how this test suite is set up, does it create an AWS bucket in the background? Seems like it. 

I would suggest using https://github.com/johannesboyne/gofakes3 for a testing API but I would like to check with @klizhentas first :-).

Reply:
@klizhentas I have added the TestSuite for third party s3, please review it. 

> I've changed the title of the PR to indicate that it's a work in progress to avoid merging for now.
> 
> By default, Teleport will enable encryption for any S3 buckets that it creates on AWS. It doesn't include any encryption of its own - it relies on encryption at the storage level. If your third party S3-compatible provider also provides encryption at rest, it should be fairly easy to enable that too.

The commit also has a new switch to disable ServerSideEncryption, as there are some providers that do not support that. Also, I did not manage to have the tests for the 3rd party usage in the same s3handler_test.go file, so I opted to use a separate one.

Test Output:
```
=== RUN   TestS3ThirdParty
2019/10/15 13:00:16 INFO HEAD BUCKET teleport-test-5358bf35-5497-4425-89ff-c69530e636c3
2019/10/15 13:00:16 INFO bucketname: teleport-test-5358bf35-5497-4425-89ff-c69530e636c3
2019/10/15 13:00:16 INFO CREATE BUCKET: teleport-test-5358bf35-5497-4425-89ff-c69530e636c3
2019/10/15 13:00:16 INFO PUT VERSIONING: Enabled
2019/10/15 13:00:16 INFO CREATE OBJECT: teleport-test-5358bf35-5497-4425-89ff-c69530e636c3 test/350e8284-161b-4fe4-82fd-0bb1d4be0a7a.tar
2019/10/15 13:00:16 INFO CREATED VERSION: teleport-test-5358bf35-5497-4425-89ff-c69530e636c3 test/350e8284-161b-4fe4-82fd-0bb1d4be0a7a.tar 3/60O30C1G60O30C1G60O30C1G60O30C1G60O30C1G60O30C1H00NOD62MJ6G7VI0=
2019/10/15 13:00:16 INFO GET OBJECT
2019/10/15 13:00:16 INFO Bucket: teleport-test-5358bf35-5497-4425-89ff-c69530e636c3
2019/10/15 13:00:16 INFO └── Object: test/350e8284-161b-4fe4-82fd-0bb1d4be0a7a.tar
OK: 2 passed
--- PASS: TestS3ThirdParty (0.01s)
PASS

Process finished with exit code 0
```